src/vmm/src/vmm_config/snapshot.rs

```diff
@@ -30,7 +30,17 @@ pub enum MemBackendType {
     /// Guest memory contents will be loaded from a file.
     File,
     /// Guest memory will be served through UFFD by a separate process.
+    /// Starting from Linux 5.11, this mode will only serve page
+    /// faults that occur in user-space, not those that occur in the kernel.
+    /// Use UffdPrivileged if you wish to serve page faults in both.
+    /// Before Linux 5.11, there is no difference between Uffd and UffdPrivileged.
+    /// Both will serve page faults in both user-space and the kernel.
```

Comment:
This is perhaps a naive question but some basic Googling and Kernel Doc skimming didn't help me. Do you have any docs that would explain what happens if I opt for Uffd (non-priviledged) while running on a Linux Kernel > 5.11 and the kernel originates a page fault? The uffd man page says that would trigger a SIGBUS, which seems like it would lead to process death/exit. So, I'm either missing something or such Kernel originated page faults are exceedingly rare in normal operation or Uffd mode will rarely (never) be used post 5.11 Kernel?

Reply:
You are right @avirtuos. You're not missing anything. We want to handle the case of kernel-triggered page faults at all times, otherwise we will get SIGBUS and die. That's what we plan to do is to use the new interface for creating UFFDs, which always gives us file descriptors that are able to handle those.